File: 0252-meeting-rooms/0252-meeting-rooms.py
class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        intervals.sort(key=lambda x: x[0])

        for i in range(len(intervals)-1):
            m1, m2=intervals[i],intervals[i+1]
            if m1[1]>m2[0]:
                return False 

        return True

File: 0252-meeting-rooms/0252-meeting-rooms.java
class Solution {
    public boolean canAttendMeetings(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[1], b[1]));
        int[] lastSchedule=new int[]{-1,-1};
        for(int i=0;i<intervals.length;i++){
            int[] currentSchedule=intervals[i];
            if(currentSchedule[0]<lastSchedule[1]){
                return false;
            }
            lastSchedule=currentSchedule;
        }
        
        return true;
    }
}